### DISCUSSION TITLE
Concourse running build monitoring

### DISCUSSION THREAD

**Discussion by bigstinky86** [concourse/concourse#7676] [General]
Hi,

We are using concourse as a CICD tool and a couple of times we found hanging builds. Every time it was the same gradle task with no errors in task logs. One of the tasks were running for 1 month. To catch this in the future we'd like to add a running build time into monitoring but after investigation of the datadogs metrics and concourse traces which are emitted we found out that they contains build time only for the finished builds but not for running. The only way we found is using API requests.

Could someone recommend some tool or implementation we could use to catch the described situations?

**Comment by pn-santos** [comment]
This https://github.com/concourse/concourse/pull/7426 (will be in the next release) will make that a non-issue since it will be possible to set a default max upper limit for tasks. Until then AFAIK there's no way to track length of time of a running build.
